subsys/net/lib/config/init.c

```diff
@@ -153,13 +163,13 @@ static void setup_ipv4(struct net_if *iface)
 		net_mgmt_add_event_callback(&mgmt4_cb);
 	}
 
-	if (sizeof(CONFIG_NET_CONFIG_MY_IPV4_ADDR) == 1) {
+	if (sizeof(ipv4_addr) == 1) {
```

Comment:
You cannot use `sizeof()` for runtime string length checking because sizeof is done at compile time. This check will always return the size of the pointer.